pkg/apis/middleware/scope.go

```diff
@@ -22,3 +35,19 @@ type RequestScope struct {
 	// it was loaded or not.
 	SessionRevalidated bool
 }
+
+// GetRequestScope returns the current request scope from the given request
+func GetRequestScope(req *http.Request) *RequestScope {
```

Comment:
I needed to move this out of `pkg/middleware` to remove circular import issues. Alot of the actual middleware uses the `requestutil.GetRequestHost` (and the like) helpers.  And now those helpers are aware of the `RequestScope` and call `GetRequestScope`